metadata_fetcher/fetchers/pra_fetcher.py

```diff
@@ -0,0 +1,159 @@
+        self.internal_collection_id = re.search(r"(?<=SO_)[0-9a-f-]+",
+                                                params.get("harvest_data").get("url")).group(0)
+        self.original_url = f"{self.BASE_URL}/structural-objects/{self.internal_collection_id}/children"
+        self.next_url = self.get_first_page_url()
+
+    def get_first_page_url(self):
+        """
+        Two possibilities exist:
+
+        1) The `original_url` contains a list of IO children and is the first page of results, or
+        2) The `original_url` is a list of SO children, hopefully contain 1 item, and we must do more
+           to get to the list of IO children:
+
+           Fetching the first page of items requires two requests to get to it. The first, the original_url,
+           returns a ChildrenResponse, from which the URL from the first Child is extracted. The second request returns
+           an EntityResponse, from which the URL is extracted from AdditionalInformation/Children's text node.
+        """
+        request = self.build_url_request(self.original_url)
+        response = requests.get(**request)
+        root = ElementTree.fromstring(response.text)
+
+        # If we have IO (Information Object) children, then this is the first page. Otherwise,
+        # we have to continue digging.
+        io_children = root.findall(".//pra:Child[@type='IO']", self.NAMESPACES)
+        if len(io_children) > 0:
+            return self.original_url
+
+        child_url = root.find(".//pra:Child[@type='SO']", self.NAMESPACES).text
+
+        request = self.build_url_request(child_url)
+        response = requests.get(**request)
+        root = ElementTree.fromstring(response.text)
+
+        return root.find(".//pra:AdditionalInformation/pra:Children", self.NAMESPACES).text
+
+    def build_fetch_request(self):
+        request = self.build_url_request(self.next_url)
+
+        print(
+            f"[{self.collection_id}]: Fetching page {self.write_page} "
+            f"at {request.get('url')}")
+
+        return request
+
+    def get_text_from_response(self, response):
+        # Starting with a list of `information-objects` URLs
+        object_url_elements = ElementTree.fromstring(response.text).findall("pra:Children/"
+                                                                            "pra:Child", self.NAMESPACES)
+
+        object_urls = [element.text for element in object_url_elements]
+
+        # Getting an individual `information-object`, extracting the URL
+        # for the oai_dc metadata fragment
+        metadata_urls = {object_url: self.get_metadata_url_from_object(object_url)
+                         for object_url in object_urls}
+
+        # Getting the metadata
+        items = {object_url: self.get_metadata_from_url(metadata_url)
+                 for (object_url, metadata_url) in metadata_urls.items()
+                 if metadata_url is not None}
```

Comment:
Here I'm gathering the metadata into a single result file, starting from the response from the request for paginated items.